targets/csharp/source/source/Extensions/Models/PlayFabCloudScriptModelsExtensions.cs.ejs

```diff
@@ -17,55 +24,222 @@ namespace PlayFab.CloudScriptModels
+    /// Provides  fields through ApiSettings and AuthenticationContext, the profile of the caller, as well as the 
+    /// arguments passed to the execution request.
+    /// </summary>
+    /// <typeparam name="TFunctionArgument">Type of FunctionArgument</typeparam>
+    public class FunctionPlayerPlayStreamContext<TFunctionArgument>
+    {
+        public PlayFabApiSettings ApiSettings { get; private set; }
+
+        public PlayFabAuthenticationContext AuthenticationContext { get; set; }
+
+        public PlayStreamEventEnvelope PlayStreamEventEnvelope { get; set; }
+
+        public ServerModels.PlayerProfile PlayerProfile { get; set; }
+
+        public TFunctionArgument FunctionArgument { get; set; }
+
+        public string CurrentPlayerId { get; set; }
 
-        private class TitleAuthenticationContext
+        protected FunctionPlayerPlayStreamContext() { }
+
+        /// <summary>
+        /// Creates a new <c>FunctionPlayerPlayStreamContext</c> out of the incoming request to an Azure Function.
+        /// </summary>
+        /// <param name="request">The request incoming to an Azure Function from the PlayFab server</param>
+        /// <returns>A new populated <c>FunctionPlayerPlayStreamContext</c> instance</returns>
+        public static async Task<FunctionPlayerPlayStreamContext<TFunctionArgument>> Create(HttpRequestMessage request)
         {
-            public string Id { get; set; }
-            public string EntityToken { get; set; }
+            using (var content = request.Content)
+            {
+                var body = await content.ReadAsStringAsync();
+                var contextInternal = PlayFabSimpleJson.DeserializeObject<FunctionPlayerPlayStreamContextInternal>(body);
+                var settings = new PlayFabApiSettings
+                {
+                    TitleId = contextInternal.TitleAuthenticationContext.Id,
+                    DeveloperSecretKey = Environment.GetEnvironmentVariable("PLAYFAB_DEV_SECRET_KEY", EnvironmentVariableTarget.Process)
+                };
+
+                var authContext = new PlayFabAuthenticationContext
+                {
+                    EntityToken = contextInternal.TitleAuthenticationContext.EntityToken
+                };
+
+                return new FunctionPlayerPlayStreamContext<TFunctionArgument>()
+                {
+                    ApiSettings = settings,
+                    AuthenticationContext = authContext,
+                    PlayStreamEventEnvelope = contextInternal.PlayStreamEventEnvelope,
+                    PlayerProfile = contextInternal.PlayerProfile,
+                    FunctionArgument = contextInternal.FunctionArgument,
+                    CurrentPlayerId = contextInternal.PlayerProfile.PlayerId
+                };
+            }
         }
+
+        private class FunctionPlayerPlayStreamContextInternal
+        {
+            public TFunctionArgument FunctionArgument { get; set; }
```

Comment:
Regarding your usage of TFunctionArgument
You have utilized reflection and generics at the same time.
You should be aware that this will not work on all platforms.
In particular, this will fail spectacularly on Unity if you try to use the same code there. Some Json serializers will fail here as well, on some platforms.
(Generally iOS)

Reply:
I'm not sure how I'm utilizing reflection here could you please explain further?
I simply added a generic type to the class and utilized that type in a field. The idea here is the customer must have a type for their `FunctionArguments` field that's in the payload to the AF, and hence can get a context that already deserializes their JSON into that type rather than deserializing it into object and them having to manually do it again. 
Also, as per our previous conversation, the platform issue you're bringing up is the reason why this class is #if'd with NETSTANDARD2_0 as it will only be applicable in Azure Functions runtime which runs on NET Standard 2+ and hence won't be compiled anywhere else. Furthermore, this is all moving to another project, as per our other conversation, and so hence we don't need to resolve this comment by fixing the code in this PR, but I'd still like to get some feedback from you regarding how I'm using reflection here.